src/main/java/com/faforever/client/leaderboard/LeaderboardService.java

```diff
@@ -179,6 +179,13 @@ public CompletableFuture<Optional<LeagueEntryBean>> getActiveLeagueEntryForPlaye
             .findFirst());
   }
 
+  public CompletableFuture<Optional<LeagueEntryBean>> getActiveLeagueEntryForPlayer(PlayerBean player, String leaderboardName) {
+    return getActiveLeagueEntriesForPlayer(player)
+        .thenApply(leagueEntries -> leagueEntries.stream()
+            .filter(leagueEntry -> Objects.equals(leagueEntry.getLeagueSeason().getLeaderboard().getTechnicalName(), leaderboardName))
+            .findFirst());
+  }
```

Comment:
Can't we filter by leaderboard in the query? That is what this call should do rather than filter java side

Reply:
I changed it, but why is that better?